### PR TITLE
MAINT: Simplify array creation routines in ndimage.

### PR DESCRIPTION
**Pull request by jaimefrio**
Gets rid of unused functionality for array creation in `nd_image.c`.

Might be easier to convince oneself of correctness looking at the PR one commit at a time.

### CODE REVIEW

File: scipy/ndimage/src/nd_image.c [hunk 1]
```diff
@@ -297,24 +284,29 @@ static PyObject *Py_Correlate1D(PyObject *obj, PyObject *args)
...
-                         &mode, &cval,
-                         NI_ObjectToLongSequence, &origin))
+                          &mode, &cval,
+                          PyArray_IntpConverter, &origin)) {
         goto exit;
+    }
+    if (!_validate_origin(input, origin)) {
+        goto exit;
+    }
     if (!NI_Correlate(input, weights, output, (NI_ExtendMode)mode, cval,
```
Comment (perimosocordiae) [inline scipy/ndimage/src/nd_image.c]: While we're cleaning things up: this condition seems to be unnecessary.

**Review by perimosocordiae — APPROVED** [review]
It's the amazing shrinking codebase! Thanks for the tip about reviewing each commit in order, that helped a lot.

File: scipy/ndimage/src/nd_image.c (hunk 1 above)
Comment (jaimefrio) [inline scipy/ndimage/src/nd_image.c]: NIce catch! There were actually many of those, have removed all I found in the latest commit.

**Comment by perimosocordiae** [comment]
Merged, to keep the ndimage work rolling.

**Pull request merged**